lib/scholar/naive_bayes/bernoulli.ex

```diff
@@ -0,0 +1,407 @@
+        provided.
+    * `:feature_log_probability` - Empirical log probability of features
+        given a class, ``P(x_i|y)``.
+  ## Examples
+      iex> x = Nx.iota({4, 3})
+      iex> y = Nx.tensor([1, 2, 0, 2])
+      iex> Scholar.NaiveBayes.Bernoulli.fit(x, y, num_classes: 3, binarize: 1.0)
+      %Scholar.NaiveBayes.Bernoulli{
+            feature_count: Nx.tensor(
+              [
+                [1.0, 1.0, 1.0],
+                [0.0, 0.0, 1.0],
+                [2.0, 2.0, 2.0]
+              ]
+            ),
+            class_count: Nx.tensor(
+              [1.0, 1.0, 2.0]
+            ),
+            class_log_priors: Nx.tensor(
+              [-1.3862943649291992, -1.3862943649291992, -0.6931471824645996]
+            ),
+            feature_log_probability: Nx.tensor(
+              [
+                [-0.40546512603759766, -0.40546512603759766, -0.40546512603759766],
+                [-1.0986123085021973, -1.0986123085021973, -0.40546512603759766],
+                [-0.28768205642700195, -0.28768205642700195, -0.28768205642700195]
+              ]
+            )
+          }
+      iex> x = Nx.iota({4, 3})
+      iex> y = Nx.tensor([1, 2, 0, 2])
+      iex> Scholar.NaiveBayes.Bernoulli.fit(x, y, num_classes: 3, force_alpha: false, alpha: 0.0)
+      %Scholar.NaiveBayes.Bernoulli{
+            feature_count: Nx.tensor(
+              [
+                [1.0, 1.0, 1.0],
+                [0.0, 1.0, 1.0],
+                [2.0, 2.0, 2.0]
+              ]
+            ),
+            class_count: Nx.tensor(
+              [1.0, 1.0, 2.0]
+            ),
+            class_log_priors: Nx.tensor(
+              [-1.3862943649291992, -1.3862943649291992, -0.6931471824645996]
+            ),
+            feature_log_probability: Nx.tensor(
+              [
+                [0.0, 0.0, 0.0],
+                [-23.025850296020508, 0.0, 0.0],
+                [0.0, 0.0, 0.0]
+              ]
+            )
+          }
+  """
+  deftransform fit(x, y, opts \\ []) do
+    if Nx.rank(x) != 2 do
+      raise ArgumentError,
+            """
+            expected x to have shape {num_samples, num_features}, \
+            got tensor with shape: #{inspect(Nx.shape(x))}\
+            """
+    end
+
+    if Nx.rank(y) != 1 do
+      raise ArgumentError,
+            """
+            expected y to have shape {num_samples}, \
+            got tensor with shape: #{inspect(Nx.shape(y))}\
+            """
+    end
+
+    {num_samples, num_features} = Nx.shape(x)
+
+    if num_samples != Nx.axis_size(y, 0) do
+      raise ArgumentError,
+            """
+            expected first dimension of x and y to be of same size, \
+            got: #{num_samples} and #{Nx.axis_size(y, 0)}\
+            """
+    end
+
+    opts = NimbleOptions.validate!(opts, @opts_schema)
+    type = to_float_type(x)
+
+    x_binarize =
+      if opts[:binarize] != nil,
+        do: Scholar.Preprocessing.Binarizer.fit_transform(x, threshold: opts[:binarize]),
+        else: x
```

Comment:
Try to move as much code as possible inside `fit_n`. For example, could this be moved there?

Reply:
I believe `opts[:binarize] != nil` inside `defn` will want to treat those as tensors. Perhaps this may work:

```elixir
case opts[:binarize] do
  nil -> x
  binarize ->Scholar.Preprocessing.Binarizer.fit_transform(x, threshold: binarize)
end
```

If it doesn't work we can leave it as is :)

Reply:
it worked, Thanks!